### PR TITLE
fix: timestamp in future for arbi exporters

### PR DESCRIPTION
**Pull request by BobTheBuidler**

### CODE REVIEW

**Review by mariuspod — APPROVED** [review]
hey, that's a good catch!
Should we make this `wait_for=True` the default ?
I've seen most places are using it now with `True`

Also there's a small improvement maybe to wait maybe `200-300ms` before retrying in the loop ?

File: yearn/utils.py [hunk 1]
```diff
@@ -57,7 +57,17 @@ def get_block_timestamp(height):
...
+def closest_block_after_timestamp(timestamp: int, wait_for: bool = False) -> int:
+    """
+    Set `wait_for = True` to make this work for future `timestamp` values.
+    """
+
+    while wait_for:
+        try:
+            return closest_block_after_timestamp(timestamp)
+        except IndexError:
+            pass
```
Comment (mariuspod) [inline yearn/utils.py:69]: wait here maybe a small time e.g. 200ms before passing on and retrying

**Comment by BobTheBuidler** [comment]
> hey, that's a good catch! Should we make this `wait_for=True` the default ? I've seen most places are using it now with `True`

I was thinking to leave all non-broken implementations of this function as-is so the behavior can't change from what is currently expected behavior. The only potential downside I could see with default value True would be an infinite loop that would be hard to identify from a debugging PoV if we ever have sync issues with a node.

File: yearn/utils.py (hunk 1 above)
Comment (BobTheBuidler) [reply yearn/utils.py:69]: I agree with this recommendation and will implement the change now.

Comment (BobTheBuidler) [reply yearn/utils.py]: Fix pushed

**Pull request merged**